statement.go

```diff
@@ -175,6 +177,71 @@ func (s *stmt) Exec(args []driver.Value) (driver.Result, error) {
 	return &result{ra}, nil
 }
 
+func (s *stmt) ExecContext(ctx context.Context, nargs []driver.NamedValue) (driver.Result, error) {
+	if s.closed {
+		panic("database/sql/driver: misuse of duckdb driver: ExecContext after Close")
+	}
+	if s.rows {
+		panic("database/sql/driver: misuse of duckdb driver: ExecContext with active Rows")
+	}
+
+	args, err := namedValueToValue(nargs)
+	if err != nil {
+		return nil, err
+	}
+	err = s.start(args)
+	if err != nil {
+		return nil, err
+	}
+
+	var pendingRes C.duckdb_pending_result
+	if state := C.duckdb_pending_prepared(*s.stmt, &pendingRes); state == C.DuckDBError {
+		dbErr := C.GoString(C.duckdb_pending_error(pendingRes))
+		C.duckdb_destroy_pending(&pendingRes)
+		C.duckdb_destroy_prepare(s.stmt)
+		return nil, errors.New(dbErr)
+	}
+	defer C.duckdb_destroy_pending(&pendingRes)
+
+	for {
+		select {
+		case <-ctx.Done():
+			return nil, ctx.Err()
+		default:
+			// continue
+		}
+		state := C.duckdb_pending_execute_task(pendingRes)
+		ready := false
+		switch state {
+		case C.DUCKDB_PENDING_RESULT_READY:
+			// we are done processing the query, now get the results
+			ready = true
+		case C.DUCKDB_PENDING_ERROR:
+			dbErr := C.GoString(C.duckdb_pending_error(pendingRes))
+			return nil, errors.New(dbErr)
+		case C.DUCKDB_PENDING_RESULT_NOT_READY:
+			// we are not done yet, continue to next task
+		default:
+			panic(fmt.Sprintf("found unkonwn state while pending execute: %T", state))
+		}
+		if ready {
+			break
+		}
```

Comment:
Maybe remove this, move `ready := false` to before the for loop, and make the loop condition `for !ready {`

Reply:
Same applies for `QueryContext`